tests/cases/compiler/declarationEmitTopLevelNodeFromCrossFile.ts

```diff
@@ -0,0 +1,16 @@
+// @declaration: true
+// @filename: a.ts
+export type X = string;
+export const fn = { o: (a?: (X | undefined)[]) => {} };
+
+// @filename: b.ts
+import {fn} from "./a";
+export const m = {                        
+    /**
+    * leading doc for prop
+    */ 
+    prop: 1
+}
+
+
+export const x = { p: fn }; 
```

Comment:
This test doesn't seem like it tests #58698; am I missing something?

Reply:
Hm, maybe I'm just wrong given the issue was about pulling comments from other files, yeah.